erpnext/accounts/doctype/accounts_settings/accounts_settings.py

```diff
@@ -12,6 +12,7 @@
 class AccountsSettings(Document):
 	def on_update(self):
 		frappe.db.set_default("auto_accounting_for_stock", self.auto_accounting_for_stock)
+		frappe.db.set_default("time_sheet_for_sales_invoice", self.time_sheet_for_sales_invoice)
 
```

Comment:
Have require sys_defaults.time_sheet_for_sales_invoice in JS side and    
"depends_on": "eval:cint(sys_defaults.time_sheet_for_sales_invoice)" in JSON